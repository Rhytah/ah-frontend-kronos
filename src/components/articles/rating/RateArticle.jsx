/* eslint-disable jsx-a11y/mouse-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import '../../../assets/scss/Landing.scss';

export default function RateArticle({ handleRating, handleMouseOver }) {
  return (
    <div className="ratings-div">
      <span id="star1" onClick={handleRating(1)} onMouseOver={handleMouseOver(1)} className="fa fa-star rating-star rate-star" />
      <span id="star2" onClick={handleRating(2)} onMouseOver={handleMouseOver(2)} className="fa fa-star rating-star rate-star" />
      <span id="star3" onClick={handleRating(3)} onMouseOver={handleMouseOver(3)} className="fa fa-star rating-star rate-star" />
      <span id="star4" onClick={handleRating(4)} onMouseOver={handleMouseOver(4)} className="fa fa-star rating-star rate-star" />
      <span id="star5" onClick={handleRating(5)} onMouseOver={handleMouseOver(5)} className="fa fa-star rating-star rate-star" />
    </div>
  );
}

RateArticle.propTypes = {
  handleRating: PropTypes.func,
  handleMouseOver: PropTypes.func,
};

RateArticle.defaultProps = {
  handleRating: () => {},
  handleMouseOver: () => {},
};
