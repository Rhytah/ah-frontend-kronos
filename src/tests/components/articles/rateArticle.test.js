/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import RateArticle from '../../../components/articles/rating/RateArticle';

describe('Component: RateArticle', () => {
  test('MyComponent renders with default props', () => {
    const wrapper = shallow(<RateArticle />);
    expect(wrapper).toMatchSnapshot();
  });
});
