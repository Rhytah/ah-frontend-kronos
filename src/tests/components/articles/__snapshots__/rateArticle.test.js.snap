// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RateArticle MyComponent renders with default props 1`] = `ShallowWrapper {}`;
